### PR TITLE
Soft disable deflate filter and add profiling regions to hdf5 output

### PR DESCRIPTION
**Pull request by pgrete**
## PR Summary

Came across this while tracking filesystem performance issues on Frontier.
Might be helpful more generally.

## PR Checklist

- [x] Code passes cpplint
- [x] New features are documented.
- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [ ] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [x] Docs build
- [ ] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

**Comment by pgrete** [comment]
Sample output from a test run (with restart output and single prec output):
```
TOP-DOWN TIME TREE:
<average time> <percent of total time> <percent time in Kokkos> <percent MPI imbalance> <remainder> <kernels per second> <number of calls> <name> [type]
=================== 
|-> 1.16e+02 sec 87.7% 0.0% 0.0% 0.8% 2.16e-01 2 MakeOutputs [region]
|   |-> 6.58e+01 sec 49.9% 0.0% 0.3% 0.3% 0.00e+00 2 PHDF5::WriteOutputFileRealPrec [region]
|   |   |-> 6.55e+01 sec 49.7% 0.0% 0.0% 0.6% 0.00e+00 2 write all variable data [region]
|   |   |   |-> 6.51e+01 sec 49.4% 0.0% 0.3% 0.1% 0.00e+00 2 write variable loop [region]
|   |   |       |-> 6.47e+01 sec 49.1% 0.0% 0.4% 100.0% 0.00e+00 2 write variable data [region]
|   |   |       |-> 3.54e-01 sec 0.3% 0.0% 8.1% 100.0% 0.00e+00 2 fill host output buffer [region]
|   |-> 4.88e+01 sec 37.0% 0.0% 0.2% 0.6% 0.00e+00 2 PHDF5::WriteOutputFileSinglePrec [region]
|   |   |-> 4.85e+01 sec 36.8% 0.0% 0.0% 0.4% 0.00e+00 2 write all variable data [region]
|   |   |   |-> 4.83e+01 sec 36.6% 0.0% 0.2% 0.1% 0.00e+00 4 write variable loop [region]
|   |   |       |-> 4.79e+01 sec 36.3% 0.0% 0.2% 100.0% 0.00e+00 4 write variable data [region]
|   |   |       |-> 3.67e-01 sec 0.3% 0.0% 6.5% 100.0% 0.00e+00 4 fill host output buffer [region]
|-> 1.55e+01 sec 11.7% 87.2% 0.0% 0.2% 2.61e+02 1 Driver_Main [region]
|   |-> 1.53e+01 sec 11.6% 88.4% 1.0% 3.1% 2.65e+02 104 MultiStage_Step [region]
```

**Review by c-prather — APPROVED** [review]
LGTM.  I'll be glad to have these regions pretty soon.

File: src/outputs/parthenon_hdf5.cpp [hunk 1]
```diff
@@ -448,13 +470,18 @@ void PHDF5Output::WriteOutputFileImpl(Mesh *pm, ParameterInput *pin, SimTime *tm
 
 #ifndef PARTHENON_DISABLE_HDF5_COMPRESSION
     PARTHENON_HDF5_CHECK(H5Pset_chunk(pl_dcreate, ndim, chunk_size.data()));
-    PARTHENON_HDF5_CHECK(
-        H5Pset_deflate(pl_dcreate, std::min(9, output_params.hdf5_compression_level)));
+    // Do not run the pipeline if compression is soft disabled.
+    // By default data would still be passed, which may result in slower output.
```
Comment (Yurlungur) [inline src/outputs/parthenon_hdf5.cpp:474]: :+1:

**Review by Yurlungur — APPROVED** [review]
Thanks for adding the instrumentation too!

**Pull request merged**